src/middleware/auth_middleware.go

```diff
@@ -0,0 +1,48 @@
+package middleware
+
+import (
+	"github.com/gin-gonic/gin"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"net/http"
+	"os"
+	"strings"
+)
+
+func TokenAuthMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		token := c.GetHeader("Authorization")
+		if token == "" {
+			c.JSON(http.StatusUnauthorized, gin.H{"error": "Authorization token not provided"})
+			c.Abort()
+			return
+		}
+
+		tokenParts := strings.Split(token, " ")
+		if len(tokenParts) != 2 || tokenParts[0] != "Bearer" {
+			c.JSON(http.StatusUnauthorized, gin.H{"error": "Invalid authentication token"})
+			c.Abort()
+			return
+		}
+
+		token = tokenParts[1]
+		cr := &rest.Config{
+			BearerToken: token,
+			Host:        os.Getenv("KUBE_API_SERVER"),
+			TLSClientConfig: rest.TLSClientConfig{
+				Insecure: true,
+			},
+		}
+
+		client, err := kubernetes.NewForConfig(cr)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to create Kubernetes client"})
+			c.Abort()
+			return
+		}
+
+		// Store the client in the context
+		c.Set("k8sClient", client)
```

Comment:
Please name it as `kubernetesClient` rather than having `8` number, also just `client` can work as well.

Reply:
I think `k8sClient` is short and indicative, and I prefer it over `kubernetesClient`. Actually, we are planning in the future to support more clients (GitLab, for instance), so renaming it to just `client` wouldn't be a good practice. What do you think?

Reply:
Ah yes, `client` is a bad idea. What about `kubeClient`? It's shorter than `kubernetesClient`.